app/models/miq_search.rb

```diff
@@ -41,6 +41,21 @@ def self.visible_to_all
     where("search_type=? or (search_type=? and (search_key is null or search_key<>?))", "global", "default", "_hidden_")
   end
 
+  def self.visible_to_current_user
+    where(:search_type => 'user', :search_key => User.current_user.userid)
+  end
+
+  def self.filters_by_type(type)
+    case type
+    when "global" # Global filters
+      visible_to_all
+    when "my"     # My filters
+      visible_to_current_user
+    else
+      raise "Error: #{type} is not a proper type!"
```

Comment:
I am thinking about this line. Do we need an error message? 

If we do then it should probably say "filter type" not just "type".